pkg/cmd/admin/policy/psp_review.go

```diff
@@ -0,0 +1,302 @@
+	$ %[1]s -u bob -g projectAdmin -f myresource.yaml
+
+	# Check whether ServiceAccount specified in podTemplateSpec in myresourcewithsa.yaml can create the Pod
+	$  %[1]s -f myresourcewithsa.yaml `)
+)
+
+const SubjectReviewRecommendedName = "subject-review"
+
+type subjectReviewOptions struct {
+	subjectReviewClient     client.PodSecurityPolicySubjectReviewsNamespacer
+	selfSubjectReviewClient client.PodSecurityPolicySelfSubjectReviewsNamespacer
+	Namespace               string
+	Filename                string
+
+	PodTemplateSpec *kapi.PodTemplateSpec
+	User            string
+	Groups          []string
+	ServiceAccount  string
+
+	PrintObject func(obj runtime.Object, out io.Writer) error
+}
+
+func (o *subjectReviewOptions) Run(out io.Writer) error {
+	var (
+		response runtime.Object
+		err      error
+	)
+	userOrSA := o.User
+	if len(o.ServiceAccount) > 0 {
+		userOrSA = o.ServiceAccount
+	}
+	switch {
+	case len(userOrSA) > 0 || len(o.Groups) > 0:
+		podSecurityPolicySubjectReview := &securityapi.PodSecurityPolicySubjectReview{
+			Spec: securityapi.PodSecurityPolicySubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+				User:     userOrSA,
+				Groups:   o.Groups,
+			},
+		}
+		response, err = o.subjectReviewClient.PodSecurityPolicySubjectReviews(o.Namespace).Create(podSecurityPolicySubjectReview)
+		if err != nil {
+			return err
+		}
+	default:
+		podSecurityPolicySelfSubjectReview := &securityapi.PodSecurityPolicySelfSubjectReview{
+			Spec: securityapi.PodSecurityPolicySelfSubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+			},
+		}
+		response, err = o.selfSubjectReviewClient.PodSecurityPolicySelfSubjectReviews(o.Namespace).Create(podSecurityPolicySelfSubjectReview)
+		if err != nil {
+			return err
+		}
+	}
+	return o.PrintObject(response, out)
+}
+
+func (o *subjectReviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command) error {
+	if len(o.User) > 0 && len(o.ServiceAccount) > 0 {
+		return fmt.Errorf("--user and --serviceaccount are mutually exclusive")
+	}
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Namespace = namespace
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.subjectReviewClient = oclient
+	o.selfSubjectReviewClient = oclient
+
+	o.Namespace = namespace
+	mapper, typer := f.Object()
+	infos, err := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &resource.FilenameOptions{Recursive: false, Filenames: []string{o.Filename}}).
+		SingleResourceType().
+		Do().
+		Infos()
+
+	if err != nil {
+		return fmt.Errorf("unable to find a resource: %v", err)
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("expected a resource for path: %s", o.Filename)
+	}
+	o.PodTemplateSpec, err = returnPodTemplateSpec(infos[0].Object)
```

Comment:
You might want to use [GetPodSpec](https://github.com/openshift/origin/blob/283a990246d914e2c2a5b00b75effedb8792b9b9/pkg/api/meta/pods.go#L52) here.